docs/src/content/docs/book/assembly-functions.mdx

```diff
@@ -0,0 +1,678 @@
+    // but do not have the optional padding
+    b{0101}
+
+    // Number literals, represented by Int values on TVM
+    42 -13
+
+    // TVM control registers
+    c0 // c0, c1, ..., c15
+
+    // TVM stack registers
+    s0 // s0, s1, ..., s255
+
+    // TVM instructions themselves
+    MYCODE // without wrapping in double-quotes "..."!
+}
+```
+
+:::caution
+
+  The `i s()` syntax for referring to stack registers beyond the $0 - 15$ range is deprecated and recognized as an error in Tact 1.6 and onward. Whenever you see `[ii] s()` in the [TVM instructions list][tvm-instructions], use one of `s0`, `s1`, ..., `s255` instead.
+
+  Additionally, in Tact 1.6 and onward, the `B{...} B>boc` syntax is deprecated in favor of `c{...}` — whenever in the [TVM instructions list][tvm-instructions] you see an instruction that accepts `[ref]` argument, such as the [`PUSHREF`](https://docs.ton.org/v3/documentation/tvm/instructions#88), use `c{...}` instead.
+
+:::
+
+## Stack calling conventions {#conventions}
+
+The syntax for parameters and returns is the same as for other function kinds, but there is one caveat — argument values are pushed to the stack before the function body is executed, and return type is what's captured from the stack afterward.
+
+### Parameters {#conventions-parameters}
+
+The first parameter is pushed to the stack first, the second one second, and so on, so that the first parameter is at the bottom of the stack and the last one at the top.
+
+```tact
+asm extends fun storeCoins(self: Builder, value: Int): Builder {
+    //                     ↑              ↑
+    //                     |              Pushed last, sits on top of the stack
+    //                     Pushed first, sits on the bottom of the stack
+
+    // Stores the value of type `Int as coins` into the Builder,
+    // taking the Builder from the bottom of the stack
+    // and Int from the top of the stack,
+    // producing a new Builder back
+    STVARUINT16
+}
+```
+
+Since the bodies of `asm{:tact}` functions do not contain Tact statements, any direct references to parameters in function bodies will be recognized as [TVM][tvm] instructions, which can easily lead to very obscure error messages.
+
+```tact
+/// Simply returns back the value of `x`
+asm fun identity(x: Int): Int { }
+
+/// COMPILATION ERROR!
+/// The `BOC` is not recognized as a parameter,
+/// but instead is interpreted as a non-existent TVM instruction
+asm fun bocchiThe(BOC: Cell): Cell { BOC }
+```
+
+The parameters of arbitrary [Struct][struct] types are distributed over their fields, recursively flattened as the arguments are pushed onto the stack. In particular, the value of the first field of the [Struct][struct] is pushed first, the second is pushed second, and so on, so that the value of the first field is at the bottom of the stack and the value of the last is at the top. If there are nested structures inside those [Structs][struct], they're flattened in the same manner.
```

Comment:
Add a warning saying this is not guaranteed and subject to change in future Tact releases

Reply:
Do we not want that behavior for Structs? This behavior seems rather convenient.

To do something else, we'd need revisit the limit on number of parameters and, perhaps, tweak Structs into Tuples underneath, which will break compatibility with FunC code. And I think we'd need to preserve the ability to import it even when we'll have the new backend — there's a lot of FunC code and it's a nice escape hatch for doing all kinds of hacks on the side (like invoking Fift in FunC's asm functions or working with continuations a bit easier).

Currently, tuples are a totally different thing barely related to structures, even emulating them via the `Struct?` hack doesn't work in all contexts — [local bindings don't type check](https://github.com/tact-lang/tact/issues/1127#issuecomment-2549296989).

Reply:
> Add a warning saying this is not guaranteed and subject to change in future Tact releases

Added a note right after that paragraph